arangod/Graph/SingleServerEdgeCursor.cpp

```diff
@@ -175,53 +180,66 @@ bool SingleServerEdgeCursor::next(EdgeCursor::Callback const& callback) {
   auto* cursorSet = &_cursors[_currentCursor];
 
   // get current cursor
-  auto cursor = (*cursorSet)[_currentSubCursor].get();
+  auto cursor = (*cursorSet)[_currentSubCursor].cursor.get();
+  uint16_t coveringPosition =
+      (*cursorSet)[_currentSubCursor].coveringIndexPosition;
+  TRI_ASSERT(cursor != nullptr);
 
   // NOTE: We cannot clear the cache,
-  // because the cursor expect's it to be filled.
+  // because the cursor expects it to be filled.
   do {
     if (cursorSet->empty() || !cursor->hasMore()) {
       if (!advanceCursor(cursor, cursorSet)) {
         return false;
       }
     } else {
-      if (cursor->hasExtra()) {
+      if (coveringPosition != aql::Projections::kNoCoveringIndexPosition) {
```

Comment:
Moving that statement into a seperate function would help to maintain readability here. (From a technical point it is not required, but IMHO it would help improve the readability. Just by reading that comparison (without further knowledge) it is hard to understand (e.g. new developers).